src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,147 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ARRAY_BEGINNING = 0;
+    private static final int EMPTY_SIZE = 0;
+
+    private int size;
+    private T[] elementStore;
+
+    ArrayList() {
+        size = EMPTY_SIZE;
+        elementStore = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementStore[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (indexIsLegal(index) || index == size) {
+            size++;
+            growIfArrayFull();
+            System.arraycopy(elementStore, index, elementStore,
+                    nextIndex(index), calculateElementsToCopy(index));
+            set(value, index);
+        } else {
+            throw createErrorWithIndex(index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
if you add to list with size 5 list with size 40 it will grow 4 times. You may calculate what new capacity you need and make it once. growToCapacity(this.size + list.size) 
Also each time it will check is it full